tests/helper/helper_generic.go

```diff
@@ -2,18 +2,25 @@ package helper
 
 import (
 	"math/rand"
+	"strings"
 	"time"
 )
 
 func init() {
 	rand.Seed(time.Now().UTC().UnixNano())
 }
 
-func randString(n int) string {
+func RandString(n int) string {
 	const letterBytes = "abcdefghijklmnopqrstuvwxyz"
 	b := make([]byte, n)
 	for i := range b {
 		b[i] = letterBytes[rand.Intn(len(letterBytes))]
 	}
 	return string(b)
 }
+
+func CmdParser(cmd string) (string, []string) {
+	cmdProgram := strings.SplitN(cmd, " ", 2)[0]
```

Comment:
I don't think that this is wise.
What if I have `cmd` that has an argument that contains space?
For example `grep "find this"`? It will break.
This is one of the reasons why we should keep passing arguments as an array , and also why original `CmdRunner` had args as a variable length slice

Reply:
> I don't think that this is wise.
What if I have cmd that has an argument that contains space?

Yes it fails if extra space is found. My intention was here not to change the command execution experience as we have been using it over a long period of time but the fact is that it has certain limitation as you mentioned. 

My suggestion is lets document the limitation in test suite and follow the old experience of writing test command. WDYT ? Even i am ok with your suggestion too :)

Reply:
Then let's have two functions, one that does it properly that will be used for new stuff, and the old one that will be there for the old tests.
We shouldn't change new tests that are using it correctly to the old way that is wrong.
If we don't start fixing it now. it will never be fixed.

Reply:
> Then let's have two functions, one that does it properly that will be used for new stuff, and the old one that will be there for the old tests.

-1, It will increase complexity and maintenance cost. 

> We shouldn't change new tests that are using it correctly to the old way that is wrong.
If we don't start fixing it now. it will never be fixed.

Right, even this was my concern too. Now i feel it's better to follow the execution format like done in core_beta_flow.

Reply:
> > Then let's have two functions, one that does it properly that will be used for new stuff, and the old one that will be there for the old tests.
> 
> -1, It will increase complexity and maintenance cost.
> 
> >

We can't continue writing new tests in a way that we know is not right.